agent/agent.go

```diff
@@ -717,7 +718,6 @@ func (a *Agent) setupClientAutoEncryptWatching(rootsReq *structs.DCSpecificReque
 			// this point, auto_encrypt was just being setup
 			// successfully.
 			autoLogger := a.logger.Named(logging.AutoEncrypt)
-			interval := a.tlsConfigurator.AutoEncryptCertNotAfter().Sub(time.Now().Add(10 * time.Second))
```

Comment:
This is the issue that staticcheck missed and `ineffassign` caught. In the error cases below `interval` is set to `1m`, but this assignment was masking the value. Moving the assignment before the loop should fix the problem.

I guess this means that auto-encrypt was polling faster than 1min in the error case?

Reply:
Yeah - this looks like a genuine bug with interval being overwritten every loop. Good catch!

Reply:
nice catch! Could you move the comments as well? I think they should be above the interval definition.

Reply:
Ah, good idea! I'm going to also move the `autoLogger` assignment above the for loop as well. I believe we can use the same one for each iteration.